tests/lms/services/canvas_api_test.py

```diff
@@ -95,22 +107,50 @@ def test_it_raises_CanvasAPIServerError_if_it_receives_an_error_response(
             exc_info.value.response == response
         ), "It passes the Canvas API response to CanvasAPIError so that it gets logged"
 
+    def test_it_validates_the_response_from_Canvas(
+        self,
+        CanvasAccessTokenResponseSchema,
+        canvas_access_token_response_schema,
+        canvas_api_client,
+        requests,
+    ):
+        response = requests.Session.return_value.send.return_value
+
+        canvas_api_client.get_token("test_authorization_code")
+
+        CanvasAccessTokenResponseSchema.assert_called_once_with(response)
+        canvas_access_token_response_schema.parse.assert_called_once_with()
+
+    def test_it_raises_CanvasAPIServerError_if_the_response_from_Canvas_is_invalid(
+        self, canvas_access_token_response_schema, canvas_api_client
+    ):
+        validation_error = ValidationError("Invalid")
+        canvas_access_token_response_schema.parse.side_effect = validation_error
+
+        with pytest.raises(CanvasAPIServerError, match="200 OK") as exc_info:
```

Comment:
I need to fix the string representation of `CanvasAPIServerError` here -- `"200 OK"` is the response status, but the response body failed validation, and the validation error message is missing from `CanvasAPIServerError` string representation.